packages/cli/src/transforms/toktx.ts

```diff
@@ -197,54 +213,83 @@ export const toktx = function (options: ETC1SOptions | UASTCOptions): Transform
 					return;
 				}
 
-				const image = texture.getImage();
-				const size = options.resize || texture.getSize();
-				if (!image || !size) {
+				let srcImage = texture.getImage()!;
+				let srcExtension = texture.getURI()
+					? FileUtils.extension(texture.getURI())
+					: ImageUtils.mimeTypeToExtension(texture.getMimeType());
+				const srcSize = texture.getSize();
+				const srcBytes = srcImage ? srcImage.byteLength : null;
+
+				if (!srcImage || !srcSize || !srcBytes) {
 					logger.warn(`${prefix}: Skipping, unreadable texture.`);
 					return;
 				}
 
+				// RESIZE: Resize textures using Sharp. KTX Software --width and --height
+				// flags apply only for raw texture creation.
+				// https://github.com/donmccurdy/glTF-Transform/issues/1348
+				//
+				// Minimum size on any dimension is 4px.
+				// https://github.com/donmccurdy/glTF-Transform/issues/502
+
+				if (options.resize || !isMultipleOfFour(srcSize[0]) || !isMultipleOfFour(srcSize[1])) {
+					const limitInputPixels = options.limitInputPixels;
+					const encoder = options.encoder as typeof sharp;
+					const instance = encoder(srcImage, { limitInputPixels }).toFormat('png');
+					const srcSize = ImageUtils.getSize(srcImage, srcMimeType)!;
+					const dstSize = options.resize
+						? Array.isArray(options.resize)
+							? fitWithin(srcSize, options.resize)
+							: fitPowerOfTwo(srcSize, options.resize)
+						: srcSize.slice();
+
+					dstSize[0] = isMultipleOfFour(dstSize[0]) ? dstSize[0] : ceilMultipleOfFour(dstSize[0]);
+					dstSize[1] = isMultipleOfFour(dstSize[1]) ? dstSize[1] : ceilMultipleOfFour(dstSize[1]);
+
+					logger.debug(`${prefix}: Resizing ${srcSize.join('x')} → ${dstSize.join('x')}px`);
+					instance.resize(dstSize[0], dstSize[1], {
+						withoutEnlargement: Array.isArray(options.resize),
+						fit: Array.isArray(options.resize) ? 'inside' : 'fill',
+						kernel: options.resizeFilter,
+					});
```

Comment:
Copied from the textureCompress implementation, this block will need changes to ensure that the output dimensions are multiples of four. Currently the `withoutEnlargement` and `fit` options may interfere with that goal.

Reply:
Continued in https://github.com/donmccurdy/glTF-Transform/pull/1378.